Tests/StreamChatUITests/Utils/CooldownTracker_Tests.swift

```diff
@@ -0,0 +1,37 @@
+//
+// Copyright © 2022 Stream.io Inc. All rights reserved.
+//
+
+import StreamChatTestHelpers
+@testable import StreamChatUI
+import XCTest
+
+final class CooldownTracker_Tests: XCTestCase {
+    var sut: CooldownTracker!
+
+    override func setUp() {
+        super.setUp()
+        sut = CooldownTracker(timer: Timer_Mock.self)
+    }
+
+    override func tearDown() {
+        sut = nil
+        super.tearDown()
+    }
+    
+    func test_onChangeIsExecutedProperly_whenCooldownTrackerIsStarts() {
```

Comment:
Currently, this is more an integration test, than a unit test, especially because we are trying to mimic the Timer behaviour in the Timer_Mock. These types of tests can be a bit flaky.

As a first glance I can image these scenarios we need to unit test:
```swift
test_start_thenScheduleTimer_andFireTimer()
test_start_whenCooldownIsZero_thenDontScheduleTimer_andDontFireTimer()
test_start_whenDurationChanges_thenOnChangeIsCalled()
test_start_whenDurationChanges_andDurationIsZero_thenInvalidateTimer()
test_start_whenDurationChanges_andDurationIsNotZero_thenDecreaseDuration_andInvalidateIsNotCalled()
```